pkg/network/msg.go

```diff
@@ -2,66 +2,60 @@ package network
 
-// RegisterRequestMsg represents the incoming register device request message
-type RegisterRequestMsg struct {
+// DeviceRegisterRequest represents the incoming register device request message
+type DeviceRegisterRequest struct {
 	ID   string `json:"id"`
 	Name string `json:"name"`
 }
 
-// UnregisterRequestMsg represents the incoming unregister device request message
-type UnregisterRequestMsg struct {
-	ID string `json:"id"`
-}
-
-// UpdateSchemaRequestMsg represents the update schema request message
-type UpdateSchemaRequestMsg struct {
-	ID     string            `json:"id"`
-	Schema []entities.Schema `json:"schema,omitempty"`
-}
-
-// RegisterResponseMsg represents the outgoing register device response message
-type RegisterResponseMsg struct {
+// DeviceRegisteredResponse represents the outgoing register device response message
+type DeviceRegisteredResponse struct {
 	ID    string  `json:"id"`
 	Token string  `json:"token"`
 	Error *string `json:"error"`
 }
 
-// UnregisterResponseMsg represents the outgoing unregister device response message
-type UnregisterResponseMsg struct {
+// DeviceUnregisterRequest represents the incoming unregister device request message
+type DeviceUnregisterRequest struct {
+	ID string `json:"id"`
+}
+
+// DeviceUnregisteredResponse represents the outgoing unregister device response message
+type DeviceUnregisteredResponse struct {
 	ID    string  `json:"id"`
 	Error *string `json:"error"`
 }
 
-// UpdateSchemaRequest represents the update schema request message
-type UpdateSchemaRequest struct {
+// SchemaUpdateRequest represents the incoming update schema request message
+type SchemaUpdateRequest struct {
 	ID     string            `json:"id"`
 	Schema []entities.Schema `json:"schema,omitempty"`
 }
 
-// UpdatedSchemaResponse represents the update schema response message
-type UpdatedSchemaResponse struct {
+// SchemaUpdatedResponse represents the outgoing update schema response message
+type SchemaUpdatedResponse struct {
 	ID string `json:"id"`
 }
 
-// ListThingsResponse represents the list things response
-type ListThingsResponse struct {
-	Things []*entities.Thing `json:"things"`
-}
-
-// RequestDataCommand represents the request data command
-type RequestDataCommand struct {
-	ID        string `json:"id"`
-	SensorIds []int  `json:"sensorIds"`
-}
-
-// AuthThingCommand represents the auth device command
-type AuthThingCommand struct {
+// DeviceAuthRequest represents the incoming auth device command
+type DeviceAuthRequest struct {
 	ID    string `json:"id"`
 	Token string `json:"token"`
 }
 
-// AuthThingResponse represents the auth device command response
-type AuthThingResponse struct {
+// DeviceAuthResponse represents the outgoing auth device command response
+type DeviceAuthResponse struct {
 	ID     string  `json:"id"`
 	ErrMsg *string `json:"error"`
 }
+
+// DeviceListResponse represents the outgoing list devices command response
+type DeviceListResponse struct {
```

Comment:
Could you add a `DeviceListRequest` struct message?

Reply:
I think we don't need to add an unused structure, just notice the [ListDevices](https://github.com/CESARBR/knot-babeltower/blob/master/pkg/thing/controllers/thing.go#L66) controller doesn't even need to deserialize the JSON message into a structure.

Reply:
I know we can understand this file as documentation and it is, but it still doesn't justify declare an unused structure. It continues being the documentation of the messages the controller layer use to deserialize the message body and extract the necessary parameters to pass to the connector.

Reply:
I understand and agree see with both point of views @esdrasjnr and @joaoaneto.

Even though I'm not fully aware of the cost of creating an empty structs, I am more inclined to go with @esdrasjnr. Simply because by doing so, we are able to group all types, errors and possibilities in general of specific things (such as message types); thus, the contributor will be able to quickly associate a part of the text documentation with the code itself.

This being said, it's something I personally struggle with (for this reason alone, I might be taking a biased P.O.V.) and have little knowledge of the cost its associated with.

Maybe @vitbaq and @ramonhpr (he is pretty busy, tho) could help us, as the result of this might (or not) become a standard coding practice in order to aid or avoid other discussions like this one.

Reply:
Good point, @lcbm. I agree with you with respect to seeking a readability improvement. I think there is no huge cost associated with leaving that structure there, in terms of **performance**. However, IMO, we need to understand that are different documentation levels and we will never reach a fully match between the high-level API documentation and the code itself.

The fact we're discussing to decide if a chunk of unused code should remain or not in the codebase is a bad sign. Maybe we should emphasize, in the API documentation, the device list command doesn't expect a payload to be deserialized. What do you think?

What really concerns me is that unused code is always bad and we shouldn't open windows to things like that.

Reply:
> _"The fact we're discussing to decide if a chunk of unused code should remain or not in the codebase is a bad sign."_

> _"What really concerns me is that unused code isn't good in general and we shouldn't open windows for that become common."_

Nice point with these two statements. As a result of this conversation, we should emphasize in the API documentation the commands which don't need payload deserialization :smile: